### PR TITLE
MAGN-8514 Add Copy/Paste to the context menu

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

If a pasted selection has groups it has an issue with copy/paste:
- create a node, then a group from the node and select the group, then copy and paste the selection anywhere => pasted group and node inside it visually are selected, but `Selection` contains only the group.

Also the PR contains some code improvement.

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 1]
```diff
@@ -559,56 +559,38 @@ internal bool CanNodeToCode(object parameters)
...
                 if (IsInRegion(region, n, fullyEnclosed))
                 {
-                    if (!DynamoSelection.Instance.Selection.Contains(n))
-                        DynamoSelection.Instance.Selection.Add(n);
+                    selection.AddUnique(n);
+                    // if annotation is selected its children should be added to selection too
+                    foreach (var m in n.SelectedModels)
+                    {
+                        selection.AddUnique(m);
+                    }
```
Comment (marimano) [inline src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs]: selection
In this case `Point` node visually selected but `Selection` doesn't contain it

File: src/DynamoCore/Models/DynamoModel.cs [hunk 2]
```diff
@@ -1751,10 +1755,9 @@ public void Paste(Point2D targetPoint)
         public void AddToSelection(object parameters)
         {
             var selectable = parameters as ISelectable;
-            if ((selectable != null) && !selectable.IsSelected)
```
Comment (marimano) [inline src/DynamoCore/Models/DynamoModel.cs]: Items inside a group has `IsSelected=true`, so they would be added

File: src/DynamoCore/Core/DynamoSelection.cs [hunk 3]
```diff
@@ -125,6 +125,18 @@ public SmartCollection(List<T> list)
...
+        /// Adds an item only if the sequence does not have it yet
+        /// </summary>
+        /// <param name="item">Item to add</param>
+        public void AddUnique(T item)
+        {
+            if (!Contains(item))
+            {
+                Add(item);
+            }
+        }
```
Comment (marimano) [inline src/DynamoCore/Core/DynamoSelection.cs]: to replace all 
```` if (!DynamoSelection.Instance.Selection.Contains(n))
                        DynamoSelection.Instance.Selection.Add(n);
````

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 4]
```diff
@@ -559,56 +559,38 @@ internal bool CanNodeToCode(object parameters)
...
-
                 if (IsInRegion(region, n, fullyEnclosed))
                 {
-                    if (!DynamoSelection.Instance.Selection.Contains(n))
-                        DynamoSelection.Instance.Selection.Add(n);
+                    selection.AddUnique(n);
                 }
                 else
                 {
                     if (n.IsSelected)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs]: please add braces for this if statement

**Comment by mjkkirschner** [comment]
@marimano thanks for finding this - one small thing, and please run it on the self serve CI, then I will merge it.

**Comment by Racel** [comment]
@marimano @mjkkirschner - when it is ready. please assign to me for review.

**Comment by marimano** [comment]
Hi @mjkkirschner, small fix is made and tests are passed:
ci

**Pull request merged**